src/operator/join.rs

```diff
@@ -570,12 +576,56 @@ where
         // timestamp to the appropriate batcher.
         output_tuples.sort_by(|(t1, _), (t2, _)| t1.cmp(t2));
 
-        let mut batch = Vec::new();
-        while !output_tuples.is_empty() {
-            let batch_time = output_tuples[0].0.clone();
+        /// Ensures that we don't leak any values if we panic while creating
```

Comment:
wouldn't it be enough to tweak the original logic to `drain` the vector from the tail? I think this way no memcpy would be happening and you wouldn't need all this complexity.

Reply:
I don't think partitioning would work the same way, the binary search it provides is a big deal perf-wise

Reply:
Right, it's silly that there is no version of `partition_point` and `binary_search_by` that work in reverse.